controllers/managedfusionoffering_controller.go

```diff
@@ -46,18 +50,33 @@ type ManagedFusionOfferingReconciler struct {
 // For more details, check Reconcile and its Result here:
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.6.4/pkg/reconcile
 func (r *ManagedFusionOfferingReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	_ = context.Background()
-	_ = r.Log.WithValues("managedfusionoffering", req.NamespacedName)
+	log := r.Log.WithValues("managedfusionoffering", req.NamespacedName)
+	log.Info("Starting reconcile for ManangedFusionOffering")
 
-	// TODO(user): your logic here
+	managedFusionOfferingList := &v1alpha1.ManagedFusionOfferingList{}
+	if err := r.Client.List(ctx, managedFusionOfferingList); err != nil {
+		return ctrl.Result{}, err
+	}
+	if len(managedFusionOfferingList.Items) == 0 {
+		r.Log.Info("no managedFusionOffering CR found")
+		return ctrl.Result{}, nil
+	}
+	for i := range managedFusionOfferingList.Items {
+		r.managedFusionOffering = &managedFusionOfferingList.Items[i]
+		switch r.managedFusionOffering.Spec.Kind {
+		case v1alpha1.KindDataFoundation:
+		case v1alpha1.KindDataFoundationClient:
```

Comment:
- the same format should be used in each reconcile function or we'll have a separate reconcile function for each offering?
- IOW, where'll the branch happen, in this switch statement itself or a switch in every function? I believe the former will be good.

Reply:
For now, we will have a switch in Reconcile function based on the offering kind
Each offering kind will have its own initReconciler and reconcilePhases function
Example: initReconcilerForODF and reconcilePhasesForODF will be implemented for ODF offering

Later, Ohad will introduce a plugin system where any offering can plug into this controller.